state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1104,7 +1104,9 @@ pub mod pallet {
 			destination_address: EncodedAddress,
 			tx_hash: TransactionHash,
 		) -> DispatchResult {
-			T::EnsureWitnessed::ensure_origin(origin)?;
+			// This extrinsic can either be witnessed called or called from within the runtime (for
```

Comment:
Is a bit nicer to create an inner function than do this. The fact we need a comment suggests it's confusing. Then in the solana_elections we can just do:
```rust
SolanaIngressEgress::contract_swap_request(
			swap_details.from,
			swap_details.to,
			swap_details.deposit_amount,
			swap_details.destination_address,
			swap_details.tx_hash,
		)
```
which makes more sense - we shouldn't need an origin at that point.

Reply:
will do it.